### PR TITLE
[chore] Update to latest otel core dependencies

### PR DESCRIPTION
**Pull request by atoulme**
In preparation of v0.105.0.

### CODE REVIEW

**Comment by atoulme** [comment]
We need to update jaeger to v1.59.0 to pick up its dependency update on expandconverter, or it brings the old internal/featuregates package and registers the same feature gate twice.

**Review by MovieStoreGuy — APPROVED** [review]

**Review by jpkrohling — APPROVED** [review]
What I reviewed:

- apart from the go.sum and go.mod, and the builder configs for otelcontribcol/oteltestbedcol, there are only changes to the jaegerremotesampling extension.

It's been a while since I last looked into that code, but the changes make sense to me. 

cc @frzifus and @yurishkuro, as the code owners for that component.

LGTM.

**Review by frzifus — APPROVED** [review]

File: extension/jaegerremotesampling/extension.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ type jrsExtension struct {
 
 	httpServer    component.Component
 	grpcServer    component.Component
-	samplingStore strategystore.StrategyStore
+	samplingStore samplingstrategy.Provider
```
Comment (frzifus) [inline extension/jaegerremotesampling/extension.go:27]: Its a renaming. Details https://github.com/jaegertracing/jaeger/commit/d8185ac6e8fc95770341ff7bab76a47b86c52b1f

**Review by songy23 — APPROVED** [review]

**Comment by codeboten** [comment]
> We need to update jaeger to v1.59.0 to pick up its dependency update on expandconverter, or it brings the old internal/featuregates package and registers the same feature gate twice.

Where are you seeing this? It looks like the tests are happy, is there no test to trigger that codepath?

**Comment by atoulme** [comment]
> Where are you seeing this? It looks like the tests are happy, is there no test to trigger that codepath?

The codepath is triggered by the module initialization when the gates register. Specifically, this affects the oteltestbedcol binary. You can see this in a previous run: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/9951143862/job/27490276615

The fix ended up being a mix of updating jaeger and setting a replace for the `expandconverter` package being brought in by jaeger: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34088/files#diff-1a887ca7885476eeb094d1eb68efffd582f91fd0cb737934aa206f42f215e2a2R406

**Review by crobert-1 — APPROVED** [review]

**Review by codeboten — APPROVED** [review]
Awesome, thanks for the response @atoulme, lgtm

**Pull request merged**